ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -564,8 +565,15 @@ private static Object extractKeyWhereClause(
     }
 
     final Expression other = getNonColumnRefSide(comparison);
-    final Object right = ((Literal) other).getValue();
+    if (!(other instanceof Literal)) {
+      throw new KsqlException("Ony comparison to literals is currently supported: " + comparison);
+    }
 
+    if (other instanceof NullLiteral) {
+      throw new KsqlException("Primary key columns can not be NULL: " + comparison);
+    }
```

Comment:
Catches errors on pull queries.

* first one catches comparison of ROWKEY to non-literal, e.g. a udf call, or some other expression. Previously this would blow up with class cast exception.
* second one catches comparison with NULL.  For STRING keys this previously returned some weird streams error and for non-STRING keys some confusing error about not being able to coerce a STRING to the required key type. (`NullLiteral` returns `"null"` from `getValue` for some weird reason).